File: gcj/2018practice/number_guessing.py
import sys

def main():
    lo, hi = map(int, input().split())
    hi += 1
    n = int(input())
    while lo + 1 < hi:
        mid = (lo + hi) // 2
        print(mid)
        sys.stdout.flush()
        resp = input()
        if resp == 'WRONG_ANSWER':
            sys.exit(0)

        if resp == 'CORRECT':
            return
        elif resp == 'TOO_SMALL':
            lo = mid
        else:
            hi = mid

T = int(input())
for _ in range(T):
    main()

File: kickstart/2019practice/number_guessing.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int T;
    cin >> T;
    while (T-- > 0) {
        int a, b;
        cin >> a >> b;
        int n;
        cin >> n;
        ++b;
        while (a + 1 < b) {
            int m = a + ((b - a) >> 1);
            cout << m << endl;
            cout.flush();
            string resp;
            cin >> resp;
            if (resp == "CORRECT") {
                a = b;
                break;
            } else if (resp == "TOO_SMALL") {
                a = m;
            } else {
                b = m;
            }
        }
    }
    return 0;
}
